app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -142,55 +171,95 @@ export function handlePPOMError(
 }
 
 function normalizePPOMRequest(
-  request: PPOMRequest | JsonRpcRequest,
-): PPOMRequest | JsonRpcRequest {
-  if (
-    !((req): req is PPOMRequest => req.method === METHOD_SEND_TRANSACTION)(
-      request,
-    )
-  ) {
-    return sanitizeRequest(request);
+  request: PPOMRequest,
+  controllerObject: TransactionMeta | SignatureRequest,
+): PPOMRequest {
+  let normalizedRequest = cloneDeep(request);
+
+  normalizedRequest = normalizeSignatureRequest(normalizedRequest);
+
+  normalizedRequest = normalizeTransactionRequest(
+    normalizedRequest,
+    controllerObject as TransactionMeta,
+  );
+
+  const { delegationMock, id, jsonrpc, method, origin, params } =
+    normalizedRequest;
+
+  return {
+    delegationMock,
+    id,
+    jsonrpc,
+    method,
+    origin,
+    params,
+  };
+}
+
+function normalizeTransactionRequest(
+  request: PPOMRequest,
+  transactionMeta: TransactionMeta,
+): PPOMRequest {
+  if (request.method !== MESSAGE_TYPE.ETH_SEND_TRANSACTION) {
+    return request;
   }
 
-  const transactionParams = request.params[0];
-  const normalizedParams = normalizeTransactionParams(transactionParams);
+  if (!Array.isArray(request.params) || !request.params[0]) {
+    return request;
+  }
+
+  const txParams = request.params[0] as TransactionParams;
+
+  txParams.gas = transactionMeta.txParams.gas;
+  txParams.gasPrice =
```

Comment:
isn't it a bit redundant and misleading to assign `transactionMeta.txParams.maxFeePerGas` to `gasPrice`, since we already assign it to `maxFeePerGas`?